doc/source/guide/derivatives_guide.rst

```diff
@@ -0,0 +1,155 @@
+.. _derivatives_in_depth:
+
+##############################################
+A guide to the different notions of derivative
+##############################################
+
+The concept of a derivative is one of the core concepts of mathematical analysis and is used where we want a linear approximation of a function in some point. Since the derivative is used in different contexts to mean slightly different things, this guide has been written to clarify and introduce what the different concepts mean when used in ODL.
```

Comment:
Still not so great. First sentence: the problem is the break to "we" style in the second half. Why not write "[...] analysis, and it is essential whenever a linear approximation of a function in some point is required."

Second sentence: "is used in different context to mean slightly different things" doesn't really make much sense. I guess what you mean is "the notion of derivative has different meanings in different contexts" or "the notion of derivatives depends on the context"
Last part: "has been written to clarify and introduce what the different concepts mean when used in ODL" -- not quite clear what is supposed to be expressed here. Do you mean "has been written to introduce the different derivative concepts used in ODL"?